### PR TITLE
github pages based schema browser

### PR DESCRIPTION
**Pull request by fritzm**
This is a functional replacement for the now-decommissioned statically-rendered schema browser, built on github pages for the sdm_schemas repo.

Github handles the rendering and serving of the pages, updating them automatically when commits are made to master in the sdm_schemas repo.   No additional tooling, services, workflow, or github actions are required.  The schema pages are rendered directly from the felis yamls within the repository, which are easily consumed and processed by the jekyll site generator used by github pages.

The main adds here are a top-level `_config.yml` file, and a new `browser/` subdirectory which contains the jekyll config and templates.  Schemas made visible in the browser are curated by providing a short markdown description for each desired to be shown in the `browser/` subdirectory (see `baseline.md`, `dp01.md`, and `hsc.md` in this PR for examples.)  A cleanup commit removing obsolete files ~and pushing others into a `deprecated/` subdirectory~ is also included here.

For a live example of these pages, see [https://fritzm.github.io/sdm_schemas/browser](https://fritzm.github.io/sdm_schemas/browser) which is running out of a clone of sdm_schemas where these changes are merged to master and github pages have been enabled.  If this PR is merged, turning on github pages on the sdm_schemas repo will make the schema browser appear at https://lsst.github.io/sdm_schemas/browser.

### CODE REVIEW

**Comment by ktlim** [comment]
Three quick comments:

1. The old browser showed things on a table by table basis, so it was clear what you were looking at when you were in the middle of a long table.  Is there some way of highlighting the currently-visible table in the sidebar or of having a persistent header in the right panel?
2. The old schema browser had a column for "not null".  It doesn't look like this is available from Felis.
3. The old browser had sortable columns in its table.  I'm not sure how useful that feature is, but it wouldn't hurt to include it.

File: browser/_layouts/schema.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    <link rel="stylesheet" href="{{ '/assets/css/style2.css?v=' | append: site.github.build_revision | relative_url }}">
+  </head>
+  <body>
+    <div id="container">
+      <nav id="sidebar">
+        <img id="logo" src="{{ '/browser/assets/images/rubinobs.png' | relative_url }}"/>
+        <ul>
+          <li><a href=".">Schema Browser</a></li>
+          <ul>
+            <li><a href="">{{ page.title }}</a></li>
```
Comment (ktlim) [inline browser/_layouts/schema.html:18]: Include the hash or tag/version somewhere?

File: browser/assets/css/style2.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+#container {
+	height: 100%;
+}
+
+#sidebar {
+	display: inline-block;
+	vertical-align: top;
+	height: 100%;
+	width: 25%;
```
Comment (ktlim) [inline browser/assets/css/style2.scss:28]: Is this a bit wide?  Maybe 20% and shrink the sidebar font a bit?

File: _config.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+theme: jekyll-theme-cayman
+data_dir: ./yml
+layouts_dir: ./browser/_layouts
+exclude:
+  - Gemfile
+  - /deprecated
```
Comment (ktlim) [inline _config.yml:6]: Since this is still available in git history, can it just be removed?  (Or moved somewhere else.)

**Review by ktlim — APPROVED** [review]

**Comment by ktlim** [comment]
Please make sure to select "Enforce HTTPS" in the GitHub Pages settings.

File: browser/index.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+---
+This schema browser provides a view on a curated subset of the Rubin Observatory's database schemas for
+user-facing data products (the Science Data Model, or SDM).  These pages are rendered directly from the latest
+revisions of schema definition YAML files maintained in the [sdm_schemas](https://github.com/lsst/sdm_schemas)
+github repository.
+
+Schemas available here for browsing include:
+
+<ul>
+    {%- assign schemas = site.pages | where: 'dir','/browser/' | sort: 'title' %}
```
[the remark this answers is not in the document]
Comment (fritzm) [reply browser/index.md:13]: That should be quite straightforward; will give it a try.

File: browser/_layouts/schema.html (hunk 1 above)
Comment (fritzm) [reply browser/_layouts/schema.html:18]: Yes, that is a (the most?) serious drawback of this approach: one version (master/main) only.  Something based on lsst-the-docs would be superior in this regard.  This solution is proposed as a low effort stopgap.

File: browser/assets/css/style2.scss (hunk 2 above)
Comment (fritzm) [reply browser/assets/css/style2.scss:28]: Will give that a try.

**Comment by fritzm** [comment]
> 1. The old browser showed things on a table by table basis, so it was clear what you were looking at when you were in the middle of a long table.  Is there some way of highlighting the currently-visible table in the sidebar or of having a persistent header in the right panel?

> 3. The old browser had sortable columns in its table.  I'm not sure how useful that feature is, but it wouldn't hurt to include it.

Would need to add a little JS for both of ^these, I think?  I (or other) could pursue this on a subsequent ticket.  (That said, perhaps that effort would be better spent looking into a replacement solution based on e.g. lsst-the-docs.)

> 2. The old schema browser had a column for "not null".  It doesn't look like this is available from Felis.

Hmm, that seems a significant lacuna in Felis?

**Comment by fritzm** [comment]
> Please make sure to select "Enforce HTTPS" in the GitHub Pages settings.

wilco

File: _config.yml (hunk 3 above)
Comment (fritzm) [reply _config.yml:6]: Not sparking joy?

File: ups/sdm_schemas.table [hunk 5]
```diff
@@ -1,4 +0,0 @@
-setupRequired(sconsUtils)
```
Comment (ctslater) [inline ups/sdm_schemas.table:1]: I think we still need to be able to setup this repo with eups? It's part of lsst_distrib.

Comment (fritzm) [reply ups/sdm_schemas.table:1]: Ah, okay.  I'll make sure it still "setups" before I merge.

Comment (fritzm) [reply ups/sdm_schemas.table:1]: @ktlim @ctslater Hmm...  Wonder if instead we should just remove from lsst_distrib?  There is no longer any code here...

Comment (ctslater) [reply ups/sdm_schemas.table:1]: It was added so that other packages (like ci_hsc) can verify that their products match the schemas here. RFC-672

Comment (fritzm) [reply ups/sdm_schemas.table:1]: Ah, okay, I see -- thanks!

**Comment by jonathansick** [comment]
We could deploy this to LSST the Docs instead quite easily (it's what Rubin uses instead of GitHub Pages). Just let me know if you want to do that.

**Comment by fritzm** [comment]
Thanks @jonathansick -- that is certainly probably best in the end, but I was unsure what exactly was needed.

In particular, ability to select versions rendered from different tags/branches would be pretty useful.  And, of course, being stylistically aligned with everything else is also always a plus.

Would we want to rework to some other rendering system than Jekyll for that?

**Pull request merged**